Source/View Controllers/UploadViewController.swift

```diff
@@ -50,11 +50,11 @@ class UploadViewController: NSViewController {
     @objc dynamic var saveCredentials = true
     @objc dynamic var readyForUpload = false
 
-    @objc dynamic var username: String!
-    @objc dynamic var password: String!
-    @objc dynamic var payloadName: String!
+    @objc dynamic var username: String?
+    @objc dynamic var password: String?
+    @objc dynamic var payloadName: String?
     @objc dynamic var payloadIdentifier: String! = UUID().uuidString
```

Comment:
The `.uuidString` property is non-nil, so rather than leaving it implicitly unwrapped, you could change this one to

```suggestion
    @objc dynamic var payloadIdentifier = UUID().uuidString
```